internal/native_image/builder.bzl

```diff
@@ -0,0 +1,128 @@
+"Logic to assemble `native-image` options."
+
+def _configure_static_zlib_compile(ctx, args, direct_inputs):
+    """Configure a static image compile against hermetic/static zlib.
+
+    Args:
+        ctx: Context of the Native Image rule implementation.
+        args: Args builder for the Native Image build.
+        direct_inputs: Inputs into the image build (mutable). """
```

Comment:
moved the static zlib setup code to the new `builder.bzl`, which is responsible for building `native-image` flags